src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,235 @@
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.InvertIf
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.InvertIf), Shared]
+    internal partial class InvertIfCodeRefactoringProvider : AbstractInvertIfCodeRefactoringProvider
+    {
+        private static readonly Dictionary<SyntaxKind, (SyntaxKind, SyntaxKind)> s_binaryMap =
+            new Dictionary<SyntaxKind, (SyntaxKind, SyntaxKind)>(SyntaxFacts.EqualityComparer)
+                {
+                    { SyntaxKind.EqualsExpression, (SyntaxKind.NotEqualsExpression, SyntaxKind.ExclamationEqualsToken) },
+                    { SyntaxKind.NotEqualsExpression, (SyntaxKind.EqualsExpression, SyntaxKind.EqualsEqualsToken) },
+                    { SyntaxKind.LessThanExpression, (SyntaxKind.GreaterThanOrEqualExpression, SyntaxKind.GreaterThanEqualsToken) },
+                    { SyntaxKind.LessThanOrEqualExpression, (SyntaxKind.GreaterThanExpression, SyntaxKind.GreaterThanToken) },
+                    { SyntaxKind.GreaterThanExpression, (SyntaxKind.LessThanOrEqualExpression, SyntaxKind.LessThanEqualsToken) },
+                    { SyntaxKind.GreaterThanOrEqualExpression, (SyntaxKind.LessThanExpression, SyntaxKind.LessThanToken) },
+                };
+
+        protected override SyntaxNode GetIfStatement(TextSpan textSpan, SyntaxToken token, CancellationToken cancellationToken)
+        {
+            var ifStatement = token.GetAncestor<IfStatementSyntax>();
+            if (ifStatement == null || ifStatement.Else == null)
+            {
+                return null;
+            }
+
+            var span = TextSpan.FromBounds(ifStatement.GetFirstToken().Span.Start, ifStatement.CloseParenToken.Span.End);
+            if (!span.IntersectsWith(textSpan))
+            {
+                return null;
+            }
+
+            return ifStatement;
+        }
+
+        protected override SyntaxNode GetRootWithInvertIfStatement(Workspace workspace, SemanticModel model, SyntaxNode ifStatement, CancellationToken cancellationToken)
+        {
+        var ifNode = (IfStatementSyntax)ifStatement;
+
+        // For single line statment, we swap the TrailingTrivia to preserve the single line
+        StatementSyntax newIfNodeStatement = null;
+        ElseClauseSyntax newElseStatement = null;
+        var newCondition = Negate(ifNode.Condition, model, cancellationToken);
+
+        if (ifNode.Statement.GetTrailingTrivia().Any(trivia => trivia.Kind() == SyntaxKind.EndOfLineTrivia))
+        {
+            // multiline if
+            newIfNodeStatement = ifNode.Else.Statement.Kind() == SyntaxKind.IfStatement  // if it's an else-if
```

Comment:
Should this check instead be `ifNode.Else.Statement.Kind() != SyntaxKind.BlockStatment`?